chain-impl-mockchain/src/certificate/evm_mapping.rs

```diff
@@ -127,6 +166,123 @@ impl DeserializeFromSlice for EvmMapping {
     }
 }
 
+/* RLP en/de ******************************************************************* */
+
+#[cfg(feature = "evm")]
+impl Decodable for EvmMapping {
+    fn decode(rlp: &Rlp<'_>) -> Result<Self, DecoderError> {
+        if rlp.item_count()? == 2 {
+            let account_id_h256 = decode_h256_from_u256(rlp.val_at::<U256>(0)?)?;
+            let account_key = chain_crypto::PublicKey::<crate::account::AccountAlg>::from_binary(
+                account_id_h256.as_fixed_bytes(),
+            )
+            .map_err(|_| DecoderError::Custom("invalid account identifier"))?;
+            return Ok(Self {
+                account_id: account_key.into(),
+                evm_address: rlp.val_at(1)?,
+            });
+        }
+        Err(DecoderError::Custom("invalid evm mapping type"))
+    }
+}
+
+#[cfg(feature = "evm")]
+impl Encodable for EvmMapping {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        s.begin_list(2);
+        s.append(&U256::from_big_endian(self.account_id.as_ref().as_ref()));
+        s.append(&self.evm_address);
+    }
+}
+
+/// Represents a signed mapping between a Jormungandr account and an EVM account.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SignedEvmMapping {
```

Comment:
We also should replace the current usage of the `EvmMapping` type with the `SIgnedEvmMapping` inside the certificate